app/modules/media/media_service.py

```
@@ -183,20 +269,266 @@ async def upload_image(
+
+            metadata = attachment.file_metadata or {}
+
+            # Check inline storage first
+            if metadata.get("text_storage") == "inline":
+                return metadata.get("extracted_text", "")
+
+            # Fetch from S3
+            elif metadata.get("text_storage") == "s3":
+                extracted_text_path = metadata.get("extracted_text_path")
+                if not extracted_text_path:
+                    raise MediaServiceError("Extracted text path not found in metadata")
```

Ruff (0.14.14)

[warning] Abstract `raise` to an inner function

(TRY301)